mesa/visualization/components/altair.py

```diff
@@ -53,6 +53,12 @@ def portray(g):
     has_size = "size" in all_agent_data[0]
     if has_size:
         encoding_dict["size"] = alt.Size("size", type="quantitative")
+    has_shape = "shape" in all_agent_data[0]
+    if has_shape:
+        encoding_dict["shape"] = agent_data["shape"]
+    has_fill = "fill" in all_agent_data[0]
+    if has_shape:
```

Comment:
I thin this should be
```suggestion
    if has_fill:
```